### DISCUSSION TITLE
Use `.devcontainer` from a parent directory

### DISCUSSION THREAD

**Discussion by a-gn** [devcontainers/community#104] [Q&A]
I have a monorepo containing separate projects in sub-folders. They have to be opened individually for VSCode to use their `pyproject.toml` and other configuration files. The root contains a common Dev Container specification for all projects.

It would be great if you could `Reopen in container` and VSCode would find e.g. `../../../.devcontainer` and use that, as long as there are not `.devcontainers` in closer parents. Tools like `clang-format` already do this with their configuration files.

Is that planned?

**Comment by samruddhikhandale** [comment]
Hi 👋 

Unfortunately, VSCode currently does not support automatically finding and using a `.devcontainer` configuration from parent directories. The `.devcontainer` directory or `devcontainer.json` file must be in the root of your project for VS Code to recognize it when reopening in a container.

However, you can manually specify the path to the `devcontainer.json` file in your workspace settings. Here's how you can do it:

1. Open the command palette with Ctrl+Shift+P (or Cmd+Shift+P on macOS).
2. Type "Preferences: Open Workspace Settings" and select it.
3. In the `settings.json` file that opens, add a new property "remote.containers.devContainerJson": "<path-to-devcontainer.json>".

Replace <path-to-devcontainer.json> with the relative path to your devcontainer.json file from the workspace root.

Here's an example:

```jsonc

{
    "remote.containers.devContainerJson": "../../../.devcontainer/devcontainer.json"
}

```

Remember to replace the path with the actual path to your `devcontainer.json` file. After you've done this, VS Code should use the specified `devcontainer.json` when you choose to reopen in a container.

Please note that this is a workaround and might not work perfectly in all scenarios. For instance, it won't automatically find the closest `.devcontainer `in parent directories like some tools do with their configuration files. 

Also, @a-gn as this request is VS Code specific, I'd recommend creating an issue in https://github.com/microsoft/vscode-remote-release/issues, the VS Code team can share better plans and guidance.

**Comment by a-gn** [comment]
Thanks @samruddhikhandale! I made a request [here](https://github.com/microsoft/vscode-remote-release/issues/9425).
